Fix pre-existing clang-tidy stddef.h error blocking debug builds

During implementation, `just build` and `cmake --build --preset debug` fail with `error: 'stddef.h' file not found [clang-diagnostic-error]` from clang-tidy when processing `src/version_info.cpp`. The clang-tidy invocation uses the conda sysroot headers but the pixi-installed clang-tidy cannot find the bundled `stddef.h`. This causes the default debug build to be broken for all contributors using the pixi environment. The workaround is `-DProjectAgamemnon_ENABLE_CLANG_TIDY=OFF`, but CI presumably runs with it enabled. The `cmake/StaticAnalyzers.cmake` should either pass `--extra-arg=--sysroot=...` pointing to the conda sysroot, or the clang-tidy invocation should use `--extra-arg=-isystem /path/to/pixi/envs/default/x86_64-conda-linux-gnu/sysroot/usr/include`.

_Follow-up from #50_

--- a/cmake/StaticAnalyzers.cmake
+++ b/cmake/StaticAnalyzers.cmake
@@ -4,7 +4,13 @@ option(${PROJECT_NAME}_ENABLE_CPPCHECK "Enable cppcheck" ON)
 if(${PROJECT_NAME}_ENABLE_CLANG_TIDY)
   find_program(CLANGTIDY clang-tidy)
   if(CLANGTIDY)
-    set(CMAKE_CXX_CLANG_TIDY ${CLANGTIDY} --extra-arg=-Wno-unknown-warning-option)
+    # Query GCC's built-in include directory to pass to clang-tidy
+    # This ensures clang-tidy can find sysroot headers like stddef.h
+    execute_process(
+      COMMAND ${CMAKE_CXX_COMPILER} -print-file-name=include
+      OUTPUT_VARIABLE GCC_INCLUDE_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    set(CMAKE_CXX_CLANG_TIDY ${CLANGTIDY} --extra-arg=-Wno-unknown-warning-option --extra-arg=-isystem${GCC_INCLUDE_DIR})
   else()
     message(WARNING "clang-tidy not found")
   endif()

